packages/idos-sdk-js/src/lib/grants/grants.ts

```diff
@@ -170,6 +188,58 @@ class ConnectedGrants extends Grants {
     });
   }
 
+  async shareMatchingEntry(
+    tableName: string,
+    publicFields: Partial<idOSCredential>,
+    privateFieldFilters: {
+      pick: Record<string, string>;
+      omit: Record<string, string>;
+    },
+    address: string,
+    lockedUntil: number,
+    receiverPublicKey: string,
+  ): Promise<{ grant: Grant; transactionId: string }> {
+    const allEntries = (await this.data.list(tableName)) as idOSCredential[];
+
+    const filteredEntries = allEntries.filter((entry) => {
+      const keys = Object.keys(publicFields) as (keyof idOSCredential)[];
+      for (const key of keys) {
+        if (entry[key] !== publicFields[key]) return false;
+      }
+      return true;
+    });
+
+    const filteredEntriesWithContent = await Promise.all(
+      filteredEntries.map(async (credential) => {
+        const fullCredential = await this.data.get(
+          "credentials",
+          (credential as unknown as idOSCredential).id,
+          false,
+        );
+
+        if (!fullCredential) return [];
```

Comment:
Hmm, didn't think about this data race.

Maybe we wanna turn this into a `filter_map` (or whatever TS wants us to do it)?